Adds custom binary config option to Osquery Manager integration (#5947)
elastic/docs-content: solutions/security/investigate/manage-integration.md

The [Osquery version](https://github.com/osquery/osquery/releases) available on an {{agent}} is associated to the version of Osquery Beat on the Agent. To get the latest version of Osquery Beat, [upgrade your {{agent}}](/reference/fleet/upgrade-elastic-agent.md).

## Use a custom Osquery version [osquery-custom-version]

> You can provide a custom Osquery binary to override the version bundled with {{agent}}. This allows you to run newer upstream versions of Osquery or internally customized builds.

Reviewer: ```suggestion
You can provide a custom Osquery binary to override the version bundled with {{agent}}. This allows you to run newer or custom versions of Osquery.
```
Bit simpler, think it gets the same point across

::::{warning}
This feature is **use at your own risk**. Elastic does not provide support or maintenance for custom Osquery binaries. Custom versions may introduce compatibility issues, including:

* Differences in the Osquery schema that affect queries and packs
* Breaking changes in flags or behavior
* Incompatibility with Elastic-managed configurations

> Elastic-specific Osquery extensions may not work properly with custom binaries. You are responsible for validating stability and compatibility within your environment.

Reviewer: ```suggestion
Elastic's Osquery extensions might not work properly with custom binaries. You are responsible for validating stability and compatibility within your environment.
```

{
   "install": {
      "linux": {
         "amd64": {
            "artifact_url": "https://example.org/osquery-5.22.1.linux_x86_64.tar.gz",
            "sha256": "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
         },
         "arm64": {
            "artifact_url": "https://example.org/osquery-5.22.1.linux_aarch64.tar.gz",
            "sha256": "fedcba9876543210fedcba9876543210fedcba9876543210fedcba9876543210"
         }
      }
   }
}
```

The following table lists the supported package formats for each platform:

| Platform | Architectures | Supported formats |
|----------|---------------|-------------------|
| Linux    | amd64, arm64  | `.tar.gz`         |
| macOS    | amd64, arm64  | `.pkg`            |
| Windows  | amd64, arm64  | `.msi`, `.zip`    |

Configuration options:

* `artifact_url`: (Required) The URL to the custom Osquery binary package. HTTPS is required by default.
* `sha256`: (Required) The SHA-256 checksum (64-character hex string) used to verify the downloaded artifact. Both `artifact_url` and `sha256` must be provided together.
* `allow_insecure_url`: (Optional) Defaults to `false`. Set to `true` to allow HTTP URLs (not recommended).

> * `ssl.verification_mode`: (Optional) TLS verification mode for HTTPS downloads.

Author: @raqueltabuyo @marc-gr please double check the configuration options here to ensure what I have written is correct, thank you! 🙏

> * `ssl.verification_mode`: (Optional) TLS verification mode for HTTPS downloads.

Reviewer: are the valid values here "true" and "false"? Might be worth specifying

Author: Boolean values so yes, good catch! ⭐
